portal/v2/src/ClusterDetailListComponents/Statistics/Statistics.tsx

```diff
@@ -0,0 +1,226 @@
+      },
+      defaultFontStyle: {
+        fontWeight: 500,
+      }
+    }
+    const theme = getTheme();
+
+    function GraphWrapper(lprops: { heading: string,
+      statisticsName: string,}
+    ) {
+      const [isModalOpen, { setTrue: showModal, setFalse: hideModal }] = useBoolean(false)
+      const [duration, setDuration] = useState<string>(globalDuration)
+      const [endDate, setEndDate] = useState<Date>(globalEndDate)
+      return (
+      <>
+      <Modal
+        titleAriaId={lprops.heading}
+        isOpen={isModalOpen}
+        onDismiss={hideModal}
+        isBlocking={false}
+      >
+        <Stack style={{ boxShadow: theme.effects.elevation8 }} styles={{root: { margin: "2px" }}}>
+        <ThemeProvider theme={darkTheme}>
+          <Stack horizontalAlign="stretch" horizontal /*className={classNames.iconContainer} /*style={{ boxShadow: theme.effects.elevation64 }}*/>
+            <Stack.Item grow={0.5}>
+              <GraphOptionsComponent duration={duration} setDuration={setDuration} endDate={endDate} setEndDate={setEndDate}/>
+            </Stack.Item>
+            <Stack.Item  align="center" grow={1}>
+              <Label> {lprops.heading} </Label> 
+            </Stack.Item>
+            <Stack.Item align="center">
+            <IconButton
+              iconProps={{ iconName: 'Cancel' }}
+              ariaLabel="Close popup modal"
+              onClick={hideModal}
+              styles={iconButtonStyles}
+            />
+          </Stack.Item>
+          </Stack>
+          </ThemeProvider>
+          <StatisticsWrapper currentCluster={props.currentCluster} detailPanelSelected={props.detailPanelSelected} loaded={props.loaded} statisticsName={lprops.statisticsName} duration={duration} endDate={endDate} graphHeight={500} graphWidth={1500}/>
+        </Stack>
+      </Modal>
+      <Stack style={{ boxShadow: theme.effects.elevation8 }} styles={{root: { margin: "2px" }}}>
+        <ThemeProvider theme={darkTheme}>
+          <Stack horizontalAlign="stretch" horizontal>
+            <Stack.Item  align="center">
+                <IconButton
+                  onClick={showModal}
+                  iconProps={{ iconName: "FullScreen" }}
+                  styles={iconButtonStyles}
+                />
+            </Stack.Item>
+            <Stack.Item  align="center" grow={1}>
+              <Text> {lprops.heading} </Text> 
+            </Stack.Item>
+            <Stack.Item>
+              <GraphOptionsComponent duration={duration} setDuration={setDuration} endDate={endDate} setEndDate={setEndDate}/>
+            </Stack.Item>  
+          </Stack>
+        </ThemeProvider>
+        <StatisticsWrapper currentCluster={props.currentCluster} detailPanelSelected={props.detailPanelSelected} loaded={props.loaded} statisticsName={lprops.statisticsName} duration={duration} endDate={endDate} graphHeight={200} graphWidth={740}/>
+      </Stack>
+      </>
+      )
+    }
+    
+    if (props.statisticsType === "api") {
```

Comment:
Is there anyway we can create a another function to create these objects? Seems like we a reusing code a bit here

Reply:
yup .. I will change it.